server/src/main/java/org/elasticsearch/common/geo/TriangleTreeWriter.java

```diff
@@ -69,13 +71,18 @@ public TriangleTreeWriter(Geometry geometry, CoordinateEncoder coordinateEncoder
     public void writeTo(StreamOutput out) throws IOException {
         out.writeInt(coordinateEncoder.encodeX(centroidCalculator.getX()));
         out.writeInt(coordinateEncoder.encodeY(centroidCalculator.getY()));
+        out.writeInt(extent.top);
+        out.writeVLong((long) extent.top - extent.bottom);
+        out.writeInt(extent.posRight);
```

Comment:
posRight and posLeft are always positive, so we can write these as VInt, right?

Reply:
posRight is initialised to Integer.MinValue so it is not always true. I have other ideas to compress an extent but it might make it less readable. This is something I would like to do but maybe in another iteration, wdyt?

Reply:
ah right. forgot